### PR TITLE
[Tests] Require exact match in assert_start_raises_init_eror (jnewbery)

### PR DESCRIPTION
**Pull request by maflcko**
Extracted from #12379, because the changes are important on their own.

This allows for exact testing, since the match can be specified with a strict regex. Internal details (such as exact formatting of the error message) can still be fuzzed away by regex wildcards.

### CODE REVIEW

**Comment by laanwj** [comment]
I did a similar thing in  #12302 initially and no one agreed :/

**Comment by MarcoFalke** [comment]
Hmm, I didn't notice the initial version of #12302, otherwise it would have been a Concept ACK from me

**Comment by laanwj** [comment]
Yeah doesn't matter, utACK

**Comment by jnewbery** [comment]
Tested ACK 58122736b53390a2013630e95ff760800af160e7

Should be tested on other platforms in case stderr is different. See https://github.com/bitcoin/bitcoin/pull/12379#issuecomment-367063010 for example

> I did a similar thing in #12302 initially and no one agreed :/

I think everyone agreed, but I argued that we should do the minimal fix required to get the v0.16 rc building and passing the tests.

**Comment by MarcoFalke** [comment]
@Sjors Does they still fail for you?

File: test/functional/test_framework/test_node.py [hunk 1]
```diff
@@ -188,8 +188,10 @@ def assert_start_raises_init_error(self, extra_args=None, expected_msg=None, *ar
                 if expected_msg is not None:
                     log_stderr.seek(0)
                     stderr = log_stderr.read().decode('utf-8')
-                    if re.fullmatch(expected_msg + '\n', stderr) is None:
-                        raise AssertionError('Expected message "{}" does not match stderr:\n"{}"'.format(expected_msg, stderr))
+                    if partial_match and re.search(expected_msg, stderr) is None:
+                        raise AssertionError('Expected message "{}" does not parially match stderr:\n"{}"'.format(expected_msg, stderr))
+                    elif re.fullmatch(expected_msg + '\n', stderr) is None:
```
Comment (jnewbery) [inline test/functional/test_framework/test_node.py]: This is wrong. Should be `elif not partial_match and re.fullmatch(...)`

or nest the whole thing one level deeper:

```
if partial match:
    if re.search(...) is None:
        #assert
else:
    if re.fullmatch(...) is None:
        #assert
```

**Comment by MarcoFalke** [comment]
Added a commit to make the tests pass on windows

**Comment by jnewbery** [comment]
Tested ACK fae137454adc50a4e1d448ed7219a8f5344486c9. Works for me.

**Comment by Sjors** [comment]
@MarcoFalke on MacOS 10.13.3. `wallet_encryption.py` still fails with a timeout. `wallet_multiwallet.py` and `wallet_multiwallet.py --usecli` now pass.

Running the full suite now; will update this comment if something new fails.

**Comment by laanwj** [comment]
Tested ACK fae1374

> @MarcoFalke on MacOS 10.13.3. wallet_encryption.py still fails with a timeout. 

Huh - we don't use any remotely complex regexes, so I wonder how this can cause timeout.

**Pull request merged**